modules/library/render/src/main/java/org/geotools/renderer/style/MarkFactoryListComparator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.renderer.style;
+
+import java.util.Comparator;
+import java.util.List;
+
+/** Mark factory comparator based nn a simple class name identifiers list. */
+public class MarkFactoryListComparator implements Comparator<MarkFactory> {
+
+    private final List<String> markFactoryList;
+
+    public MarkFactoryListComparator(List<String> markFactoryList) {
+        this.markFactoryList = markFactoryList;
+    }
+
+    @Override
+    public int compare(MarkFactory mf1, MarkFactory mf2) {
+        Integer index1 = getIndex(mf1);
+        Integer index2 = getIndex(mf2);
```

Comment:
No need for boxing, or am I missing something:
```java
int index1 = getIndex(mf1);
int index2 = getIndex(mf2);
return Integer.compare(index1, index2);
```

Reply:
Good point, fixing